actionmailer/lib/action_mailer/railtie.rb

```diff
@@ -45,13 +46,9 @@ class Railtie < Rails::Railtie # :nodoc:
           self.delivery_job = delivery_job.constantize
         end
 
-        if smtp_settings = options.delete(:smtp_settings)
-          self.smtp_settings = smtp_settings
-        end
-
         if smtp_timeout = options.delete(:smtp_timeout)
-          self.smtp_settings[:open_timeout] ||= smtp_timeout
-          self.smtp_settings[:read_timeout] ||= smtp_timeout
+          options.smtp_settings[:open_timeout] ||= smtp_timeout
```

Comment:
We should not be mutating the `options` here. I think we can just avoid doing a delete a dup the options before passing.